distribution/lib/Standard/Table/0.0.0-dev/src/Excel/Excel_Workbook.enso

```diff
@@ -289,12 +297,50 @@ type Excel_Workbook
 
     ## PRIVATE
     with_java_workbook : (Workbook -> Any) -> Any ! Illegal_State
-    with_java_workbook self f = Illegal_State.handle_java_exception <|
+    with_java_workbook self f = Illegal_State.handle_java_exception <| handle_invalid_location <| Empty_Sheet.handle_java_exception <|
         case self.excel_connection_resource_ref.get of
             Nothing ->
                 Error.throw (Illegal_State.Error "The workbook is being used after it was closed.")
             resource -> resource.with connection->
                 connection.withWorkbook f
 
+    ## GROUP Standard.Base.Input
+       ICON data_input
+       Reads one or more sheets into a Table.
+
+       Arguments:
+       - sheet_names: the names of the sheets to read. Defaults to all sheets.
+         An invalid `sheet_name` will be reported as a problem.
+       - headers: whether to use the first row as headers (default is
+         `Detect_Headers`) for each sheet.
+       - return: how to return the tables. Defaults to `Merged_Table` meaning
+         the tables are merged into a single table.
+       - on_problems: how to handle problems during reading. Defaults to
+         `Report_Warning`.
+    @sheet_names (s-> build_sheet_selector s)
```

Comment:
I assume there probably is a reason, but I'm surprised - what is the reason the following could not work?

```suggestion
    @sheet_names build_sheet_selector
```